src/common/prompts.go

```diff
@@ -0,0 +1,171 @@
+	if err != nil {
+		return nil, err
+	}
+	return &list[index], nil
+}
+
+func PromptForLevels(client *opslevel.Client) (*opslevel.Level, error) {
+	list, err := client.ListLevels()
+	if err != nil {
+		return nil, err
+	}
+
+	templates := &promptui.SelectTemplates{
+		Label:    "{{ . }}?",
+		Active:   fmt.Sprintf("%s {{ .Name | cyan }} ({{ .Index | red }})", promptui.IconSelect),
+		Inactive: "    {{ .Name | cyan }}",
+		Selected: fmt.Sprintf("%s {{ .Name | faint }} ({{ .Index | red }})", promptui.IconGood),
+		Details: `
+		{{ "Alias:" | faint }}	{{ .Alias }}
+		{{ "Description:" | faint }}	{{ .Description }}`,
+	}
+
+	prompt := promptui.Select{
+		Label:     "Select Level",
+		Items:     list,
+		Templates: templates,
+		Size:      len(list),
+	}
+
+	index, _, err := prompt.Run()
+	if err != nil {
+		return nil, err
+	}
+	return &list[index], nil
+}
+
+func PromptForFilter(client *opslevel.Client) (*opslevel.Filter, error) {
+	list, err := client.ListFilters()
+	if err != nil {
+		return nil, err
+	}
+
+	templates := &promptui.SelectTemplates{
+		Label:    "{{ . }}?",
+		Active:   fmt.Sprintf("%s {{ .Name | cyan }}", promptui.IconSelect),
+		Inactive: "    {{ .Name | cyan }}",
+		Selected: fmt.Sprintf("%s {{ .Name | faint }}", promptui.IconGood),
+	}
+
+	noneValue := opslevel.Filter{
+		Name: "None",
+		Id: nil,
+	}
+	list = append([]opslevel.Filter{noneValue}, list...)
+
+	prompt := promptui.Select{
+		Label:     "Select Filter",
+		Items:     list,
+		Templates: templates,
+		Size:      len(list),
+	}
+
+	index, _, err := prompt.Run()
+	if err != nil {
+		return nil, err
+	}
+	return &list[index], nil
+}
+
+func PromptForTeam(client *opslevel.Client) (*opslevel.Team, error) {
+	list, err := client.ListTeams()
+	if err != nil {
+		return nil, err
+	}
+
+	templates := &promptui.SelectTemplates{
+		Label:    "{{ . }}?",
+		Active:   fmt.Sprintf("%s {{ .Name | cyan }}", promptui.IconSelect),
+		Inactive: "    {{ .Name | cyan }}",
+		Selected: fmt.Sprintf("%s {{ .Name | faint }}", promptui.IconGood),
+		Details: `
+		{{ "Aliases:" | faint }}	{{ .Aliases }}
+		{{ "Manager:" | faint }}	{{ .Manager.Name }}`,
+	}
+
+	noneValue := opslevel.Team{
+		Name: "None",
+	}
+	list = append([]opslevel.Team{noneValue}, list...)
+
+	prompt := promptui.Select{
+		Label:     "Select Team",
+		Items:     list,
+		Templates: templates,
+		Size:      len(list),
+	}
+
+	index, _, err := prompt.Run()
+	if err != nil {
+		return nil, err
+	}
+	return &list[index], nil
+}
+
+func PromptForIntegration(client *opslevel.Client) (*opslevel.Integration, error) {
+	list, err := client.ListIntegrations()
+	if err != nil {
+		return nil, err
+	}
+
+	templates := &promptui.SelectTemplates{
+		Label:    "{{ . }}?",
+		Active:   fmt.Sprintf("%s {{ .Name | cyan }}", promptui.IconSelect),
+		Inactive: "    {{ .Name | cyan }}",
+		Selected: fmt.Sprintf("%s {{ .Name | faint }}", promptui.IconGood),
+		Details: `
+		{{ "Type:" | faint }}	{{ .Type }}`,
+	}
+
+	n := 0
```

Comment:
Same as above, open to feedback but based this on an example for updating a list in place so did the same here to filter the list

Reply:
Its bad practice to mutate a list that you are iterating over.  I would instantiate a new list and then append items to it as you loop through the original list.  This remove the need for the var `n` and the increment in the loop as well as the slice assign on 157.

Also you can combind loops and a single if statement like this

```
filteredList := []opslevel.Integration{}
for _, value := range list; val.Type == "generic" {
   filteredList = append(newList, val)
}
```